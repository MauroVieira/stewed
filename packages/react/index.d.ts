/**
 * Required to avoid errors when importing these types of files
 */

declare module "*.svg?url" {
  const content: string;
  export default content;
}

declare module "*.svg" {
  const content: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
  export default content;
}

declare module "*.md" {
  const content: string;
  export default content;
}

declare module "*.scss" {
  const classes: { [key: string]: string };
  export default classes;
}
