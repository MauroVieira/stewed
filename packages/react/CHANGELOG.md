# @stewed/react

## 0.1.0

### Minor Changes

- a302a43: Update all packages

### Patch Changes

- Updated dependencies [a302a43]
  - @stewed/utilities@0.2.0
  - @stewed/hooks@0.1.0
  - @stewed/tokens@0.2.0
