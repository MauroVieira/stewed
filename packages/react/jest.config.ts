import { jestConfig } from "@harmonix-hub/testing";

module.exports = {
  ...jestConfig(),
  testEnvironment: "jsdom",
  moduleNameMapper: {
    "\\.(css|less|scss|sss|styl)$": "<rootDir>/node_modules/jest-css-modules",
  }
};
