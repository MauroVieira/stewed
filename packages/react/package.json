{
  "name": "@stewed/react",
  "version": "0.0.0",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "dist/**"
  ],
  "scripts": {
    "dev": "npx vite",
    "build": "npx vite build",
    "lint": "TIMING=1 eslint \"src/**/*.ts*\"",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
  },
  "devDependencies": {
    "@stewed/tsconfig": "workspace:*",
    "@stewed/utils": "workspace:*",
    "@stewed/lint": "workspace:*",
    "@types/react": "^18.0.9",
    "@types/react-dom": "^18.0.4",
    "eslint": "^8.15.0",
    "react": "^18.1.0",
    "tsup": "^5.10.1",
    "typescript": "^4.5.3",
    "sass": "^1.58.0",
    "@vitejs/plugin-react": "^1.3.2",
    "vite": "^2.9.15"
  },
  "publishConfig": {
    "access": "public"
  }
}
