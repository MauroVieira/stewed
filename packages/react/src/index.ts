export * from "./accordion";
export * from "./alert";
export * from "./aspect-ratio";
export * from "./avatar";
export * from "./badge";
export * from "./box";
export * from "./button";
export * from "./card";
export * from "./carousel";
export * from "./checkbox";
export * from "./divider";
export * from "./list-box";
export * from "./progress";
export * from "./radio";
export * from "./tabs";
export * from "./tag";
export * from "./text-area";
export * from "./text-field";
export * from "./theme";
export * from "./theme";
export * from "./toggle";
export * from "./typography";

// Utilities
export * from "./types";
