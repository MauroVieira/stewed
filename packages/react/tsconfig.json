{
  "extends": "@harmonix-hub/tsconfig/react.json",
  "compilerOptions": {
    "outDir": "./dist",
  },
  "include": ["custom.d.ts", "index.ts", "./src/**/*", "./src/index.ts"],
  "exclude": ["dist", "src/**/stories", "node_modules"]
}
