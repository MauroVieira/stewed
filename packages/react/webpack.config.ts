import path from "path";
// Types
import type { Configuration } from "webpack";

const webpackConfig: Configuration = {
  mode: "production",
  entry: "./src/index.ts",
  output: {
    filename: "index.js",
    library: {
      type: "module",
    },
    path: path.resolve(__dirname, "dist"),
  },
  experiments: {
    outputModule: true,
  },
  optimization: {
    minimize: true,
  },
  module: {
    rules: [
      {
        test: /\.s[ac]ss$/i,
        use: [
          "style-loader",
          {
            loader: "css-loader",
            options: {
              modules: {
                auto: true,
                localIdentName: "[hash:base64]",
              },
              sourceMap: true,
            },
          },
          "sass-loader",
        ],
      },
      {
        test: /\.tsx?$/i,
        exclude: /node_modules/,
        use: {
          loader: "ts-loader",
          options: {
            compilerOptions: {
              sourceMap: true,
            },
          },
        },
      },
    ],
  },
  resolve: {
    extensions: [".tsx", ".ts", ".js"],
  }
};

export default webpackConfig;
