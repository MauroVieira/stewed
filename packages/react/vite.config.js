import { defineConfig } from "vite";
// Node
import { resolve } from "path";
// Plugins
import dts from "vite-plugin-dts";
import peerDepsExternal from "rollup-plugin-peer-deps-external";
import react from "@vitejs/plugin-react";
import { libInjectCss } from "vite-plugin-lib-inject-css";

export default defineConfig({
  plugins: [
    react(),
    libInjectCss(),
    dts({
      copyDtsFiles: true
    }),
  ],
  build: {
    copyPublicDir: false,
    lib: {
      entry: resolve(__dirname, "src/index.ts"),
      formats: ["es"],
    },
    rollupOptions: {
      plugins: [peerDepsExternal()],
      output: {
        assetFileNames: "assets/[name][extname]",
        entryFileNames: "[name].js",
      },
    },
  },
});
