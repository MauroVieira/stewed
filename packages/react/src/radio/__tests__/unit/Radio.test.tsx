import React from "react";
// UI Components
import { Radio, type RadioProps } from "../../index";
// Utilities
import { render, act, fireEvent } from "@testing-library/react";

describe("Radio", () => {
  describe("Snapshots", () => {
    it("should renders default component", () => {
      const { container } = render(<Radio>Radio</Radio>);

      expect(container).toMatchSnapshot();
    });

    it("should render additional classes", () => {
      const { container } = render(<Radio className="other-class">Radio</Radio>);

      expect(container).toMatchSnapshot();
    });

    it.each<RadioProps["skin"]>(["primary", "critical", "success"])(
      "should apply '%s' skin classes",
      (skin) => {
        const { container } = render(<Radio skin={skin}>Radio</Radio>);

        expect(container).toMatchSnapshot();
      }
    );

    it.each<RadioProps["appearance"]>(["border", "default"])(
      "should apply '%s' appearance classes",
      (appearance) => {
        const { container } = render(<Radio appearance={appearance}>Radio</Radio>);

        expect(container).toMatchSnapshot();
      }
    );

    it.each<RadioProps["size"]>(["sm", "md", "lg"])("should apply '%s' size classes", (size) => {
      const { container } = render(<Radio size={size}>Radio</Radio>);

      expect(container).toMatchSnapshot();
    });
  });

  describe("Events", () => {
    it("should trigger `onChange` event when value changes", () => {
      const handleChange = jest.fn();

      // Render the Radio component
      const { getByLabelText } = render(
        <Radio aria-label="radio-input" onChange={handleChange}>
          Radio
        </Radio>
      );

      // Get the radio input by its ARIA label
      const input = getByLabelText("radio-input");

      // Simulate the change event
      act(() => {
        fireEvent.change(input, { target: { checked: true } });
      });

      // Check if the input is checked
      expect((input as HTMLInputElement).checked).toBe(true);
    });
  });
});
