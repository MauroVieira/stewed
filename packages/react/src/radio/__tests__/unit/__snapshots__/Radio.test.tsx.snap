// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Snapshots should apply 'border' appearance classes 1`] = `
<div>
  <label
    class="radio radio--primary radio--border radio--md"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should apply 'critical' skin classes 1`] = `
<div>
  <label
    class="radio radio--critical radio--md"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should apply 'default' appearance classes 1`] = `
<div>
  <label
    class="radio radio--primary radio--md"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should apply 'lg' size classes 1`] = `
<div>
  <label
    class="radio radio--primary radio--lg"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should apply 'md' size classes 1`] = `
<div>
  <label
    class="radio radio--primary radio--md"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should apply 'primary' skin classes 1`] = `
<div>
  <label
    class="radio radio--primary radio--md"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should apply 'sm' size classes 1`] = `
<div>
  <label
    class="radio radio--primary radio--sm"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should apply 'success' skin classes 1`] = `
<div>
  <label
    class="radio radio--success radio--md"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should render additional classes 1`] = `
<div>
  <label
    class="radio radio--primary radio--md other-class"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;

exports[`Radio Snapshots should renders default component 1`] = `
<div>
  <label
    class="radio radio--primary radio--md"
  >
    <input
      class="radio__input"
      type="radio"
      value=""
    />
    <span
      class="radio__control"
    />
    <div
      class="radio__text"
    >
      Radio
    </div>
  </label>
</div>
`;
